coded e2e tests for http server ... debug in process

mizutombo opened the pull request:
* When running 'npm test', I am getting multiple instances of error 'app.address is not a function' within my 'http-server-e2e.js' code ... and, despite numerous debug attempts, I am unable to resolve this error.

GeoffreyEmerson commented on package.json:
```diff
@@ -33,6 +32,7 @@
     "eslint": "^3.8.1",
-    "mocha": "^3.1.2"
+    "mocha": "^3.1.2",
+    "nodemon": "^1.11.0"
```
You're missing `chai-http` from your devDependencies.

GeoffreyEmerson commented on test/http-server-e2e.js:
```diff
@@ -0,0 +1,41 @@
+const chaiHttp = require('chai-http');
+chai.use(chaiHttp);
+const assert = chai.assert;
+const server = require('../http-server');
```
Here's part of your problem. `http-server.js` is your main, which doesn't export a server. You want to require the file that exports the object made by the "createServer" call, which in this case is `httpCowSayFiglet`. Make this fix and you'll get some more useful error messages.

GeoffreyEmerson commented on test/http-server-e2e.js:
```diff
@@ -0,0 +1,41 @@
+      .get('/happy_cow') // error ... app.address is not a function
+      .end((err, res) => {
+        if (err) return done(err);
+        assert.strictEqual(res.body, 'hello wonderful world');
```
Since your server is sending back a simple string and not json, you'll find the response at 'res.text' instead of 'res.body'.

mizutombo replied:
Hi Geoff,
Based on your suggestions, I have made some changes to my code, but still keep getting the exact same error message.

GeoffreyEmerson commented:
You didn't need to change anything in your "http-server.js" or "httpCowSayFiglet.js" files. They were fine as is. You only needed to change which one you required in your test for chai to work with. Chai-http works with your server object, not the listener. It makes its own listener functionality.

mizutombo commented:
Geoff ... thanks for all your help in debugging my code.
I reverted back to the 'http-server.js' and httpCowSayFiglet.js' files, added additional server e2e tests, and validated functionality and passage of all tests.
